### PR TITLE
EN-1977 Enhance image metadata attributes (master)

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by eugeniosant** [comment]
Retest this please

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java [hunk 1]
```diff
@@ -265,13 +262,33 @@ public Element getJDOMElement() {
...
+        this.addResourceMetadatasElement(attributeElement);
         return attributeElement;
     }
 
+    /*
+    <metadatas>
+        <metadata key="xxxxx" lang="en">value</metadata>
+    </metadatas>
+     */
+    protected void addResourceMetadatasElement(Element attributeElement) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java]: This needs a simple unit test

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java [hunk 2]
```diff
@@ -45,21 +47,29 @@ public Object getAttributeHandlerPrototype() {
 
     @Override
     public void startAttribute(Attributes attributes, String qName) throws SAXException {
-        if (qName.equals("resource")) {
+        if (this.isIntoMetadatas()) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: Unit test

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java [hunk 3]
```diff
@@ -75,21 +85,35 @@ private void startResource(Attributes attributes, String qName) throws SAXExcept
 
     @Override
     public void endAttribute(String qName, StringBuffer textBuffer) {
-        if (qName.equals("resource")) {
+        if (this.isIntoMetadatas()) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: Unit tests

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java [hunk 4]
```diff
@@ -75,21 +85,35 @@ private void startResource(Attributes attributes, String qName) throws SAXExcept
...
+                || qName.equals(IResourceManager.DESCRIPTION_METADATA_KEY)
+                || qName.equals(IResourceManager.LEGEND_METADATA_KEY)
+                || qName.equals(IResourceManager.TITLE_METADATA_KEY)) {
+            this.endMetadata(qName, textBuffer);
         } else {
             super.endAttribute(qName, textBuffer);
         }
     }
 
+    private void endMetadata(StringBuffer textBuffer) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: unit test

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java [hunk 5]
```diff
@@ -38,36 +40,42 @@
...
+    private Map<String, List<String>> mapping;
+    private String metadataKey;
+    private List<String> metadataKeys;
 
     private IContentManager contentManager;
     private IResourceManager resourceManager;
     private ICmsSearchEngineManager searchEngineManager;
 
+    @Override
+    public void validate() {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java]: Unit tests

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java [hunk 6]
```diff
@@ -81,26 +89,47 @@ public String updateSystemParams() {
...
+
+    public String removeMetadata() {
+        Map<String, List<String>> newMapping = this.buildMapping();
+        if (!StringUtils.isBlank(this.getMetadataKey())) {
+            newMapping.remove(this.getMetadataKey().trim());
+        }
+        return SUCCESS;
+    }
+
+    protected Map<String, List<String>> buildMapping() {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java]: Unit tests for this one and for remove

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java (hunk 1 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java]: Done

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java (hunk 2 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: Done

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java (hunk 3 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: Done

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java (hunk 4 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/parse/extraAttribute/ResourceAttributeHandler.java]: Done

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java (hunk 5 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java]: Done

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java (hunk 6 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java]: Done

**Comment by entando-service** [comment]
All is Well

**Pull request merged**